addons/stock_account/migrations/13.0.1.1/post-migration.py

```diff
@@ -0,0 +1,225 @@
+        "write_date": price_history_rec["datetime"],
+        "stock_move_id": None,
+        "unit_cost": 0.0,
+        "remaining_qty": 0.0,
+        "remaining_value": 0.0,
+        "company_id": company.id,
+        "product_id": product.id,
+        "description": _("Product value manually modified (from %s to %s)"
+                         ) % (previous_price, price_history_rec["cost"]),
+        "value": value,
+        "quantity": 0.0,
+    }
+    return svl_vals
+
+
+def get_product_price_history(env, company_id, product_id):
+    env.cr.execute("""
+        SELECT id, company_id, product_id, datetime, cost, create_uid, write_uid, write_date
+        FROM product_price_history
+        WHERE company_id = %s AND product_id = %s
+        ORDER BY datetime, id
+    """, (company_id, product_id))
+    return env.cr.dictfetchall()
+
+
+def get_stock_moves(env, company_id, product_id):
+    env.cr.execute("""
+        SELECT sm.id, sm.company_id, sm.product_id, sm.date, sm.product_qty, sm.reference,
+            COALESCE(sm.price_unit, 0.0) AS price_unit,
+            sm.create_uid, sm.create_date, sm.write_uid, sm.write_date,
+            CASE WHEN (sl.usage <> 'internal' AND (sl.usage <> 'transit' OR sl.company_id <> sm.company_id))
+                   AND (sld.usage = 'internal' OR (sld.usage = 'transit' AND sld.company_id = sm.company_id))
+                   THEN 'in'
+                 WHEN (sl.usage = 'internal' OR (sl.usage = 'transit' AND sl.company_id = sm.company_id))
+                   AND (sld.usage <> 'internal' AND (sld.usage <> 'transit' OR sld.company_id <> sm.company_id))
+                   THEN 'out'
+                 WHEN sl.usage = 'supplier' AND sld.usage = 'customer' THEN 'dropship'
+                 WHEN sl.usage = 'customer' AND sld.usage = 'supplier' THEN 'dropship_return'
+                 ELSE 'other'
+            END AS move_type
+        FROM stock_move sm LEFT JOIN stock_location sl ON sl.id = sm.location_id
+            LEFT JOIN stock_location sld ON sld.id = sm.location_dest_id
+        WHERE sm.company_id = %s AND sm.product_id = %s
+            AND state = 'done'
+        ORDER BY sm.date, sm.id
+    """, (company_id, product_id))
+    return env.cr.dictfetchall()
+
+
+@openupgrade.logging()
+def generate_stock_valuation_layer(env):
+    company_obj = env["res.company"]
+    product_obj = env["product.product"]
+    # Needed to modify global variable
+    global precision_price
+    precision_price = env["decimal.precision"].precision_get("Product Price")
+    precision_uom = env["decimal.precision"].precision_get(
+        "Product Unit of Measure"
+    )
+    companies = company_obj.search([])
+    products = product_obj.with_context(active_test=False).search([("type", "in", ("product", "consu"))])
+    all_svl_list = []
+    for product in products:
+        for company in companies:
+            _logger.info("Product:{} {} Company:{}".format(product.id, product.name, company.name))
```

Comment:
is needed to log this? I have a bbdd with a fucking lot of variants, and this log it's just spam

Reply:
Yes, don't log on loops. I'm preparing a progressbar for not needing to know "where I am".

Reply:
Ok, remove it.
I added it while I was testing the performance, but at the current speed of the process (milliseconds) it doesn't make sense.